app/main.py

```diff
@@ -1,3 +1,58 @@
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    @classmethod
+    def checkvalue(cls, other: "Distance") -> float:
```

Comment:
You can just use type annotations and don't check for other types. It is up to developer call methods with the correct data types if there are annotations
So you can remove this method